### DISCUSSION TITLE
Is there a way to set API baseUrl and other options when using @kubb/swagger-tanstack-query ?

### DISCUSSION THREAD

**Discussion by obrassard** [kubb-labs/kubb#1187] [Q&A] [answered]
Hello ! 

I am wondering how can I set the base URL for an api when using hooks generated by `@kubb/swagger-tanstack-query`.
Also, is it possible to set default headers such as `Authorization: Bearer ....` in the underlying Axios client ?

Thanks in advance !

**Comment by stijnvanhulle** [comment, marked as answer]
You can use a custom client like described here: https://v3.kubb.dev/knowledge-base/how-tos/fetch. There you could set a base URL or you could also set `serverIndex`: https://v3.kubb.dev/documentation/plugins/plugin-oas/#serverindex.

**Reply by obrassard** [reply]
Thank you !
